File: src/main/java/de/ayont/lpc/channels/PermissionChannel.java
package de.ayont.lpc.channels;

import de.ayont.lpc.LPC;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;

public class PermissionChannel extends BaseChannel {

    public PermissionChannel(LPC plugin, String id, ConfigurationSection section) {
        super(plugin, id, section);
    }

    @Override
    public boolean canRead(Player player, Player sender) {
        return player.hasPermission(permission);
    }
}

File: src/main/kotlin/de/ayont/lpc/channels/PermissionChannel.kt
package de.ayont.lpc.channels

import de.ayont.lpc.LPC
import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player

class PermissionChannel(plugin: LPC, id: String, section: ConfigurationSection) : BaseChannel(plugin, id, section) {

    override fun canRead(player: Player, sender: Player): Boolean {
        return player.hasPermission(getPermission())
    }
}
